lib/fog/core/model.rb

```diff
@@ -21,16 +21,7 @@ def initialize(new_attributes = {})
     end
 
     def inspect
-      Thread.current[:formatador] ||= Formatador.new
-      data = "#{Thread.current[:formatador].indentation}<#{self.class.name}"
-      Thread.current[:formatador].indent do
-        unless self.class.attributes.empty?
-          data << "\n#{Thread.current[:formatador].indentation}"
-          data << self.class.attributes.map { |attribute| "#{attribute}=#{send(attribute).inspect}" }.join(",\n#{Thread.current[:formatador].indentation}")
-        end
-      end
-      data << "\n#{Thread.current[:formatador].indentation}>"
-      data
+      Fog::Formatador.new(self)
```

Comment:
I agree that the use of it is cryptic, you actually have to dig into the code to figure out what it's doing and that's not a good thing(tm).  I think what you're suggesting is probably the better way to go. I think the idea was to simply move the logic in its own container seeing as how it was taking up so much space in collection.rb and model.rb and to keep things DRY. This all began with trying to make rubocop happy, and I still think it should be appeased. So if we say Fog::Formatador.inspect(self), what about the issue of not including nested objects in the case of model.rb? What do you think the call to inspect should look like in that case? Also, the reason I made it a class and not a module was something you said about indentation getting whacky due to threads, iirc. I am much more in favor of having a module, this isn't anything that needs to keep state IMHO.